src/device-registry/utils/create-site.js

```diff
@@ -575,13 +577,19 @@ const manageSite = {
     try {
       let response = {};
       let promises = [];
+      const today = monthsInfront(0);
+      const oneMonthAgo = monthsInfront(-1);
+      const endDate = generateDateFormatWithoutHrs(today);
+      const startDate = generateDateFormatWithoutHrs(oneMonthAgo);
       const paths = constants.GET_ROAD_METADATA_PATHS;
       const arrayOfPaths = Object.entries(paths);
       for (const [key, path] of arrayOfPaths) {
         const url = constants.GET_ROAD_METADATA({
```

Comment:
modified the function to include additional parameters such that we do not have to require module while in the constants file. Hope this prevents the circular dependency. @BenjaminSsempala

Reply:
Thank you @Baalmart . But I think this might not directly solve the dependency because the dependency is still between `date.js` and `constants.js` .

Reply:
Thanks @BenjaminSsempala for this discovery! Much appreciated!

Reply:
A quick one @BenjaminSsempala ....when this date function was removed from constants and required "outside" as per those recent changes, how does a dependency still exist between `constants` and `date`?. I am somehow confused now. Perhaps I need to think deeper!

Were you able to `pull` all the changes to your local machine?

Thanks again!

Reply:
Yes, thank you @Baalmart. Not all the `date` functions were removed from `constants` . The `generateDateFormatWithoutHrs` which is imported from `date` is still used inside constants.

![image](https://user-images.githubusercontent.com/86492979/194277767-915b8b8d-83af-436b-bb9e-017087cf22a7.png)

 This is why it's only the `generateDateFormatWithoutHrs` function that is now exported from `date` before any function from `constants` is required...(to prevent `constants` from being called again before its dependency is satisfied.)

![image](https://user-images.githubusercontent.com/86492979/194278274-4ff8e8a9-ac5f-4066-b2ea-37d3bd1849eb.png)

I hope this answers your question. Thank You!

Reply:
Thank you @Baalmart. So alternatively, we could define a `generateDateFormatWithoutHrs` function inside the constants file, such that the `constants` file doesn't require anything from the `dates` file. Although wouldn't that be a bit repetitive?